docs/prow/manage-component-jobs-with-templates.md

```diff
@@ -0,0 +1,224 @@
+```
+
+5. Merge the changes.
+
+Create a PR with your changes in the `config.yaml` file and the job files generated by the Render Templates.
+
+After your PR is reviewed and approved, merge the changes to the `test-infra` repository. The job configuration is automatically applied to the Prow production cluster. The `config_updater` plugin configured in the `prow/plugins.yaml` file adds a comment to the PR:
+
+![msg](./assets/msg-updated-config.png).
+
+6. Create a Makefile for your component.
+
+Buildpacks require `Makefile` defined in your component directory under the `kyma` repository. The `Makefile` has to define the **ci-release** target that is executed for a PR issued against the release branch.
+
+See an example of `Makefile` for the Console Backend Service component that already uses the generic buildpack:
+
+```Makefile
+APP_NAME = console-backend-service
+APP_PATH = components/$(APP_NAME)
+BUILDPACK = eu.gcr.io/kyma-project/test-infra/buildpack-golang-toolbox:v20190930-d28d219
+SCRIPTS_DIR = $(realpath $(shell pwd)/../..)/scripts
+
+include $(SCRIPTS_DIR)/go-dep.mk
+
+VERIFY_IGNORE := /vendor\|/automock\|/testdata\|/pkg
+
+.PHONY: path-to-referenced-charts
+path-to-referenced-charts:
+	@echo "resources/core"
+
+```
+
+> **NOTE** Add a tab before each command.
+
+If your job involves pushing a Docker image, its name is based on the following environment variables:
+
+- **DOCKER_TAG** that refers to the Docker tag set by the `build.sh` script.
+- **DOCKER_PUSH_DIRECTORY** that points to the directory in the Docker repository where the image is pushed. Set it in the job definition by adding the **preset-build-pr**, **preset-build-master**, or **preset-build-release** Preset.
+- **DOCKER_PUSH_REPOSITORY** that is the Docker repository where the image is pushed. It is set in the job definition by the **preset-docker-push-repository** Preset.
+
+7. Change your component job and test to obligatory.
+
+Create another PR in the `test-infra` repository that removes these entries:
+
+- `optional: true` from your component job definition in `templates/config.yaml`.
+- `jobsuite.Optional()` from your component test definition in `components_test.go`.
+
+This change makes your component job and test obligatory to pass on all PRs before they can be merged.
+
+</details>
+<details>
+<summary>
+Modify component jobs
+</summary>
+
+To change component job configuration, follow these steps:
+
+1. In the `config.yaml` file, change the name of the file where the jobs are generated. For example, add the `deprecated` suffix.
+2. Add `until: {last release}` to this configuration. It specifies the release until which this component version applies.
+3. Create a new entry with the new configuration. Set the `to` field to point to the file responsible for storing jobs.
+4. Add `since: {next release}` to the new entry. It specifies the release from which this component version applies.
+
+See this example:
```

Comment:
Starting from this place I'm wondering if we shouldn't indent the text down to line 171 but I'll leave it up to you.